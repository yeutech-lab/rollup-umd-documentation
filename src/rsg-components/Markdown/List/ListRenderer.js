import React, { Children, cloneElement } from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';
import Styled from 'rsg-components/Styled';

const styles = ({ space, color, fontFamily }) => ({
	list: {
		marginTop: 0,
		marginBottom: space[2],
		paddingLeft: space[3],
		fontSize: 'inherit',
	},
	ordered: {
		listStyleType: 'decimal',
	},
	li: {
		color: color.base,
		fontFamily: fontFamily.base,
		fontSize: 'inherit',
		lineHeight: 1.5,
		listStyleType: 'inherit',
	},
});

export function ListRenderer({ classes, ordered, children }) {
	const Tag = ordered ? 'ol' : 'ul';

	const classNames = cx(classes.list, ordered && classes.ordered);

	return (
		<Tag className={classNames}>
			{Children.map(children, li => cloneElement(li, { className: classes.li }))}
		</Tag>
	);
}
ListRenderer.propTypes = {
	classes: PropTypes.object.isRequired,
	ordered: PropTypes.bool,
	children: PropTypes.node.isRequired,
};
ListRenderer.defaultProps = {
	ordered: false,
};

export default Styled(styles)(ListRenderer);
