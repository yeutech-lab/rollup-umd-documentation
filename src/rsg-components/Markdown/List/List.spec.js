import React from 'react';

import List from './index';

describe('Markdown List', () => {
	it('should render an unordered list', () => {
		const actual = render(
			<List>
				<li>First</li>
				<li>Second</li>
			</List>
		);

		expect(actual).toMatchSnapshot();
	});

	it('should render an ordered list', () => {
		const actual = render(
			<List ordered>
				<li>First</li>
				<li>Second</li>
			</List>
		);

		expect(actual).toMatchSnapshot();
	});
});
