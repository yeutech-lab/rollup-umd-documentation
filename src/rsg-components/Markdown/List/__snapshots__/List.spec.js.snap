// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Markdown List should render an ordered list 1`] = `
<ol
  class="rsg--list-0 rsg--ordered-1"
>
  <li
    class="rsg--li-2"
  >
    First
  </li>
  <li
    class="rsg--li-2"
  >
    Second
  </li>
</ol>
`;

exports[`Markdown List should render an unordered list 1`] = `
<ul
  class="rsg--list-0"
>
  <li
    class="rsg--li-2"
  >
    First
  </li>
  <li
    class="rsg--li-2"
  >
    Second
  </li>
</ul>
`;
